### DISCUSSION TITLE
MetaMask approve

### DISCUSSION THREAD

**Discussion by AranAragan** [TenKeyLabs/dappwright#17] [Q&A] [answered]
Hello!
I'm trying to close this window - via await metamask.approve(); but it doesn't work for me

    await testPage.getByRole('button', { name: 'Connect a Wallet' }).click();
    await testPage.getByRole('button', { name: 'Metamask' }).click();
    await metamask.approve();

<img width="1189" alt="image" src="https://user-images.githubusercontent.com/62331710/209670196-b3739f8d-59ec-4b55-9a4a-9ae47a4f1a5e.png">

**Comment by osis** [comment, marked as answer]
Hello!

Looked into this with a bit more context via DMs. Looks like this is an issue with a button click and not dappwright itself? Sent a working example and video.

Cheers
